src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/ChainedMethodCallWrapCheck.java

```diff
@@ -0,0 +1,496 @@
+ * <p>
+ * Example:
+ * </p>
+ * <pre>
+ * void method() {
+ *
+ *     assertWithSomeMessage("ex").that(1).isEqualTo(1);
+ *                // violation above, ''3' method calls on single line, max allowed is '1''
+ *
+ *     assertWithSomeMessage("ex")
+ *         .that(1).isEqualTo(1); // ok, 2 method calls on current line &lt;= maxCallsPerLine
+ *
+ *     assertWithSomeMessage("ex")
+ *         .that(1)
+ *         .isEqualTo(1);         // ok, every method call is wrapped
+ *
+ *     Truth.assertWithSomeMessage("ex").that(1).isEqualTo(1);
+ *                                // ok above, identifierPattern is different
+ *
+ *     Truth.assertWithSomeMessage("ex")
+ *         .that(1).isEqualTo(1); // ok above, identifierPattern is different, also
+ *                                // 2 method calls on current line &lt;= maxCallsPerLine
+ * }
+ * </pre>
+ * <p>
+ * To configure the check to use custom identifierPattern, maxSingleLinesCalls and maxCallsPerLine:
+ * </p>
+ * <pre>
+ * &lt;module name=&quot;ChainedMethodCallWrap&quot;&gt;
+ *  &lt;property name=&quot;identifierPattern&quot; value=&quot;^assert.*$&quot; /&gt;
+ *  &lt;property name=&quot;maxSingleLineCalls&quot; value=&quot;2&quot; /&gt;
+ *  &lt;property name=&quot;maxCallsPerLine&quot; value=&quot;2&quot; /&gt;
+ * &lt;/module&gt;
+ * </pre>
+ * <p>
+ * Example:
+ * </p>
+ * <pre>
+ * void method() {
+ *
+ *     assertWithSomeMessage("ex").that(1).isEqualTo(1);
+ *           // violation above, ''3' method calls on single line, max allowed is '2''
+ *
+ *     assertWithSomeMessage("ex")
+ *         .that(1).isEqualTo(1); // ok, 2 method calls on current line &lt;= maxCallsPerLine
+ *
+ *     assertWithSomeMessage("ex")
+ *         .that(1)
+ *         .isEqualTo(1);         // ok, every method call is wrapped
+ *
+ *     Truth.assertWithSomeMessage("ex").that(1).isEqualTo(1);
+ *                                // ok above, identifierPattern is different
+ *
+ *     Truth.assertWithSomeMessage("ex")
+ *         .that(1).isEqualTo(1); // ok, identifierPattern is different, also
+ *                                // 2 method calls on current line &lt;= maxCallsPerLine
+ * }
+ * </pre>
+ * <p>
+ * Parent is {@code com.puppycrawl.tools.checkstyle.TreeWalker}
+ * </p>
+ * <p>
+ * Violation Message Keys:
+ * </p>
+ * <ul>
+ * <li>
+ * {@code too.many.single.line}
+ * </li>
+ * <li>
+ * {@code wrap.method.call}
+ * </li>
+ * </ul>
+ *
+ * @since 10.1
+ */
+@StatelessCheck
+public class ChainedMethodCallWrapCheck extends AbstractCheck {
+
+    /**
+     * A key is pointing to the warning message text in "messages.properties"
+     * file.
+     */
+    public static final String MSG_TOO_MANY_SINGLE_LINE = "too.many.single.line";
+
+    /**
+     * A key is pointing to the warning message text in "messages.properties"
+     * file.
+     */
+    public static final String MSG_WRAP_METHOD_CALL = "wrap.method.call";
+
+    /**
+     * Default string regex used by identifierPattern.
```

Comment:
Please add explanation that this matches nothing.

Reply:
Done.

Reply:
Resolved.